### PR TITLE
fix build against libressl

### LINKED ISSUE #3027: LibreSSL Support
LibreSSL is a compatible fork of OpenSSL which you can use to compile Telegram Desktop with.
It is a fork by OpenBSD project, with goal of refactoring the code .It was created as a response to the Heartbleed vulnerability.
To this date, it has had less vulnerabilities compared to OpenSSL, because it has better code audit than OpenSSL.

Please provide proper compile instructions for using LibreSSL with Telegram Desktop.

### LINKED ISSUE #4415: Building problem & code removal
### Steps to reproduce
1. Building telegram-desktop with LibreSSL

### Expected behaviour
Working executable

### Actual behaviour
build fails. 

It builds if FIPS_mode_set() call is removed. It seems that `FIPS_mode_set` can be removed, since it does nothing.
### Configuration
TrueOS-Desktop-201712150936/FreeBSD 12 CURRENT, clang7
**Version of Telegram Desktop:**
1.23
**Used theme**:

<details><summary><b>Logs</b>:</summary>
```
/wrkdirs/usr/ports/net-im/telegram-desktop/work/tdesktop-1.1.23/Telegram/SourceFiles/core/utils.cpp:311:3: error: use of undeclared identifier 'FIPS_mode_set'
                FIPS_mode_set(0);
```

It builds if FIPS_mode_set() call is removed. It seems that `FIPS_mode_set` can be removed, since it does nothing.

</details>

### PR DESCRIPTION
**Pull request by Johnnynator**
resolves #4415 resolves #3027

### CODE REVIEW

File: Telegram/SourceFiles/core/utils.cpp [hunk 1]
```diff
@@ -295,7 +295,9 @@ namespace ThirdParty {
 		av_lockmgr_register(nullptr);
 
 		CRYPTO_cleanup_all_ex_data();
+#if !defined(LIBRESSL_VERSION_NUMBER)
```
Comment (stek29) [inline Telegram/SourceFiles/core/utils.cpp]: why not ifndef? :P

Comment (Johnnynator) [reply Telegram/SourceFiles/core/utils.cpp]: Just personal preference. I changed it to ifndef now.

Comment (stek29) [reply Telegram/SourceFiles/core/utils.cpp]: @Johnnynator wasn't necessary, I just was wondering :P

File: Telegram/SourceFiles/core/utils.cpp [hunk 2]
```diff
@@ -295,7 +295,9 @@ namespace ThirdParty {
 		av_lockmgr_register(nullptr);
 
 		CRYPTO_cleanup_all_ex_data();
+#ifndef IBRESSL_VERSION_NUMBER
```
Comment (xvitaly) [inline Telegram/SourceFiles/core/utils.cpp]: Typo here, I think. You need `LIBRESSL_VERSION_NUMBER`, not `IBRESSL_VERSION_NUMBER`.

**Comment by digitalist** [comment]
@Johnnynator we took this to https://github.com/trueos/freebsd-ports/pull/99/commits/93860617e7d1e28b5ca237cc706cf6ea134cf318 thank you

**Comment by john-preston** [comment]
Thanks!

**Pull request merged**